src/Faker/Generator/SeedableIntegerGenerator.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Faker\Generator;
+
+/**
+ * @experimental
+ */
+interface SeedableIntegerGenerator extends IntegerGenerator
+{
+    /**
+     * Seeds the number generator.
+     */
+    public function seed(int $value): void;
```

Comment:
Allowing the reseeding of an existing Generator is probably not a good idea. The native engines (PHP 8.2+) intentionally support constructor-seeding only to make their behavior as predictable as possible.

Reply:
> fakerphp/faker currently allows seeding, that's why I extracted a separate interface.

In this case it would probably be cleaner to completely replace the “generator” instance whenever reseeding happens. But I don't know if that's feasible with the current architecture / user expectations.

As I said over in #752, it would make sense to embrace the native OO API. As you're breaking compatibility in a new major anyway you might as well go the extra mile. YMMV.

Reply:
@TimWolla 

As you may have noticed in #752, there is quite some resistance towards even adopting a PHP version support policy, let alone dropping support for PHP versions without active support. Dropping support here for PHP 8.1 is out of the question right now.

The aim here is to

- extract an interface for generating (random) integers
- extracting an implementation of that interface that reflects the current functionality
- making room for alternative implementations of that interface

Everything that is broken in `fakerphp/faker:1.23` and `fakerphp/faker:2.0` right now by using 

- `mt_rand()` with arguments
- `mt_rand()` without arguments
- `mt_getrandmax()`
- `mt_srand()`

is totally fine for every single user of this package - they have had time to get used to it.

Again, the aim here is to move forward in small steps.

Reply:
FWIW, I would also be somewhat resistant to dropping support for a PHP version if the increased minimum requirement does not provide a value-add. Adopting the new random API would be a value-add, though.

I don't have any personal or professional interest in Faker, I no longer use it (I've contributed some stuff back in 2013 and 2014). I'm just here commenting from the perspective of one of the randomness maintainers in PHP itself, hoping it would be useful to not make avoidable mistakes in the design of the new major version (e.g. the [`mt_rand()` without parameters](https://github.com/FakerPHP/Faker/pull/769#discussion_r1331643920) one).